test/app/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/gorilla/mux"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+// simple test application
+// allows to check if users can use the provided connectionstring,
+// and if data remains after changes are made
+
+var (
+	collection *mongo.Collection
+	ctx        = context.TODO()
+
+	// related to DB
+	dbName         = "Ships"
+	collectionName = "ShipSpec"
+	port           = 8080
+)
+
+func main() {
+	r := newRouter()
+	http.ListenAndServe(fmt.Sprintf(":%d", port), r)
+}
+
+func newRouter() *mux.Router {
+	r := mux.NewRouter()
+	r.HandleFunc("/", handler).Methods("GET")
+	r.HandleFunc("/mongo/", getKeyValue).Methods("GET")
+	r.HandleFunc("/mongo/", postKeyValue).Methods("POST")
+	r.HandleFunc("/mongo/", deleteKeyValue).Methods("DELETE")
+	return r
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "It is working")
+}
+
+func getKeyValue(w http.ResponseWriter, r *http.Request) {
+	ship := getShipFromRequest(r)
+	collection, err := getMongoCollection(dbName, collectionName)
+	if err != nil {
+		panic(err)
+	}
+	filter := bson.M{"key": ship.Key}
+	var foundShip ModelShip
+	collection.FindOne(ctx, filter).Decode(&foundShip)
+	res, _ := json.Marshal(&foundShip)
+	fmt.Fprintf(w, string(res))
+}
+
+func postKeyValue(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "Lets see\n")
+	collection, err := getMongoCollection(dbName, collectionName)
+	if err != nil {
+		panic(err)
+	}
+
+	ship := getShipFromRequest(r)
+	fmt.Fprintf(w, "got ship key: " +ship.Key+"\n")
+	res, err := collection.InsertOne(ctx, ship)
+	if err != nil {
+		panic(err)
+	}
+
+	response, _ := json.Marshal(res)
+	fmt.Fprintf(w, string(response))
+}
+
+func deleteKeyValue(w http.ResponseWriter, r *http.Request) {
+	ship := getShipFromRequest(r)
```

Comment:
the same as above - I'd suggest passing the id here instead of the whole object, what do you think?